### ISSUE TITLE
"You do not have permission to post to this room" error shown for no apparent reason

### ISSUE THREAD

**Issue by tgr** [vector-im/riot-web#11741]
Steps to reproduce (tested with riot.im/app and riot.im/develop):
* use a registered nick
* join [`#mediawiki-core`](https://riot.im/app/#/room/#freenode_#mediawiki-core:matrix.org)
* try to send a message

The input field where you normally type the messages is disabled and shows `You do not have permission to post to this room`.

The channel has `+r` so presumably that's confusing Riot (or maybe the bridge). Seen from another client the join looks like `tgr_ [tgrmatrixo@wikipedia/tgr] has joined #mediawiki-core` so clearly not a real permission issue.

**Comment by tgr** [comment]
Actually, I don't seem to receive messages either. So maybe this is an issue with the appserver's own user being banned? (I do get a user list, but it contains 400+ users while actually the channel only has ~70. And I do get an event with `tgr joined and left and joined, XXX and 24 others joined, YYY left`, where the joins are users who were indeed on the channel when I joined, but they are only a subset of all users on the channel; and the leave is a real event that happened shortly after joining.)

The last old event I see in the chat history is `@appservice-irc:matrix.org kicked @tgr:matrix.org. Reason: IRC error on #mediawiki-core: err_needreggednick` but it's from almost a year ago so presumably unrelared?

**Comment by turt2live** [comment]
The bridge has decided that you need a higher power level to speak in that room. The last kick you saw is just a function of the bridge and probably unrelated indeed.

The bridge would need to promote you, though I'm not sure how/when it does that - #irc:matrix.org is in a better position to help.

**Comment by tgr** [comment]
Thanks, moved to matrix-org/matrix-appservice-irc#938.

Beyond the bridge misidentifying the power level needed the post, there is still the issue of not seeing messages by others, while the UI indicates I have joined successfully. (Which is more problematic than merely not being able to join - it looks as if I had joined successfully, and there were no new messages, when that's not actually the case.) Is that also strictly a problem with the bridge?
